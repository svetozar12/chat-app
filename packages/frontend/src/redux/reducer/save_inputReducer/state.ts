export default interface ISave_inputState {
  input_username: string;
  input_password: string;
  input_email: string;
  input_gender: string;
  notification_number: number;
}
