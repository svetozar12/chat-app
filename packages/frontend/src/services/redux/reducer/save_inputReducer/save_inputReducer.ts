import ISave_inputState from "./state";
import { HYDRATE } from "next-redux-wrapper";
import { ActionType } from "../../types";

export const initialState: ISave_inputState | ISave_inputState[] = {
  input_username: "",
  input_password: "",
  input_email: "",
  input_gender: "other",
  input_file: null,
  notification_number: 0,
};
const saveInputReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.SAVE_INPUT_USERNAME:
      return { ...state, input_username: action.payload };
    case ActionType.SAVE_INPUT_PASSWORD:
      return { ...state, input_password: action.payload };
    case ActionType.SAVE_INPUT_EMAIL:
      return { ...state, input_email: action.payload };
    case ActionType.SAVE_INPUT_GENDER:
      return { ...state, input_gender: action.payload };
    case ActionType.SAVE_INPUT_FILE:
      return { ...state, input_file: action.payload };
    case ActionType.NOTIFICATION_NUMBER:
      return { ...state, notification_number: action.payload };
    default:
      return state;
  }
};

export default saveInputReducer;
