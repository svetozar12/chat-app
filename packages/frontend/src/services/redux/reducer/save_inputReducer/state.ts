export default interface ISave_inputState {
  input_username: string;
  input_password: string;
  input_email: string;
  input_gender: "male" | "female" | "other";
  input_file: any;
  notification_number: number;
}
