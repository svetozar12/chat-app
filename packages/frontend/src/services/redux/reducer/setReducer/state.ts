export interface IInitialSet {
  reciever: string;
  chat_inviter: string;
  pageNumber: number;
  setUserSettings: boolean;
  setFriendRequest: boolean;
  toggleCreateGroup: boolean;
  setMobileNav: boolean;
  setIsMatch: boolean;
  setChatSettings: boolean;
  setModalInvite: boolean;
  setIsLoggingIn: boolean;
  setIsLoading: boolean;
}
