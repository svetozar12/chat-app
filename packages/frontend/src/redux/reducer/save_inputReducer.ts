import { Action, ActionType, InitialState3 } from "../state";
import { HYDRATE } from "next-redux-wrapper";
export const initialState = {
  input_username: "",
  input_password: "",
  input_email: "",
  input_gender: "",
  notification_number: 0,
};
const saveInputReducer = (state: InitialState3 = initialState, action: Action) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.SAVE_INPUT_USERNAME:
      return { ...state, input_username: action.payload };
    case ActionType.SAVE_INPUT_PASSWORD:
      return { ...state, input_password: action.payload };
    case ActionType.SAVE_INPUT_EMAIL:
      return { ...state, input_email: action.payload };
    case ActionType.SAVE_INPUT_GENDER:
      return { ...state, input_gender: action.payload };
    case ActionType.NOTIFICATION_NUMBER:
      return { ...state, notification_number: action.payload };
    default:
      return state;
  }
};

export default saveInputReducer;
