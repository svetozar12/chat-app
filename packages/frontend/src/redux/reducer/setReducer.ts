import { Action, ActionType, InitialState2 } from "../state";
export const initialState = {
  reciever: "",
  chat_inviter: "",
  pageNumber: 2,
  setUserSettings: false,
  setFriendRequest: false,
  toggleCreateGroup: false,
  setMobileNav: false,
  setIsMatch: false,
  setChatSettings: false,
  setModalInvite: false,
  setIsLoggingIn: false,
};
const setReducer = (state: InitialState2 | any | InitialState2[] = initialState, action: Action) => {
  switch (action.type) {
    case ActionType.SET_RECIEVER:
      return { ...state, reciever: action.payload };
    case ActionType.CHAT_INVITER:
      return { ...state, chat_inviter: action.payload };
    case ActionType.INCREMENT_PAGE_NUMBER:
      return { ...state, pageNumber: ++action.payload };
    case ActionType.SET_FRIEND_REQUEST:
      return { ...state, setFriendRequest: action.payload };
    case ActionType.SET_USER_SETTINGS:
      return { ...state, setUserSettings: action.payload };
    case ActionType.TOGGLE_CREATE_GROUP:
      return { ...state, toggleCreateGroup: action.payload };
    case ActionType.SET_MOBILE_NAV:
      return { ...state, setMobileNav: action.payload };
    case ActionType.SET_CHAT_SETTINGS:
      return { ...state, setChatSettings: action.payload };
    case ActionType.SET_IS_MATCH:
      return { ...state, setIsMatch: action.payload };
    case ActionType.SET_MODAL_INVITE:
      return { ...state, setModalInvite: action.payload };
    case ActionType.SET_IS_LOGEDIN:
      return { ...state, setIsLoggingIn: action.payload };
    case ActionType.SIGN_OUT:
      return initialState;
    default:
      return state;
  }
};

export default setReducer;
