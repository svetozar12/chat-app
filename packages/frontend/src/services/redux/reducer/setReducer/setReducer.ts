import { AnyAction } from 'redux';
import { ActionType } from '../../types';
import { IInitialSet } from './state';

export const initialState: IInitialSet | IInitialSet[] = {
  reciever: '',
  chat_inviter: '',
  pageNumber: 2,
  setUserSettings: false,
  setFriendRequest: false,
  toggleCreateGroup: false,
  setMobileNav: false,
  setIsMatch: false,
  setChatSettings: false,
  setModalInvite: false,
  setIsLoggingIn: false,
  setIsLoading: false,
};
const setReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.SET_RECIEVER:
      return { ...state, reciever: action.payload };
    case ActionType.CHAT_INVITER:
      return { ...state, chat_inviter: action.payload };
    case ActionType.INCREMENT_PAGE_NUMBER:
      return { ...state, pageNumber: ++action.payload };
    case ActionType.SET_FRIEND_REQUEST:
      return { ...state, setFriendRequest: action.payload };
    case ActionType.SET_USER_SETTINGS:
      return { ...state, setUserSettings: action.payload };
    case ActionType.TOGGLE_CREATE_GROUP:
      return { ...state, toggleCreateGroup: action.payload };
    case ActionType.SET_MOBILE_NAV:
      return { ...state, setMobileNav: action.payload };
    case ActionType.SET_CHAT_SETTINGS:
      return { ...state, setChatSettings: action.payload };
    case ActionType.SET_IS_MATCH:
      return { ...state, setIsMatch: action.payload };
    case ActionType.SET_MODAL_INVITE:
      return { ...state, setModalInvite: action.payload };
    case ActionType.SET_IS_LOGEDIN:
      return { ...state, setIsLoggingIn: action.payload };
    case ActionType.SET_IS_LOADING:
      return { ...state, setIsLoading: action.payload };
    case ActionType.SIGN_OUT:
      return initialState;
    default:
      return state;
  }
};

export default setReducer;
