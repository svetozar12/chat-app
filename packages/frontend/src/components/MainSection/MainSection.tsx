/* eslint-disable no-unused-vars */
// components
import { css, cx } from '@emotion/css';
import { CloseButton, Flex, Slide, useColorModeValue, VStack } from '@chakra-ui/react';
import ActiveChats from './ActiveChats';
import FindFriends from './FindFriends';
import ChatSettings from './ChatSettings';
// other
import useThemeColors from '../../hooks/useThemeColors';
import { connect } from 'react-redux';
import { STATE } from 'services/redux/reducer';
import { bindActionCreators, Dispatch } from 'redux';
import { toggleChatSettings } from 'services/redux/reducer/toggles/actions';
import { useGetChatListQuery } from 'services/generated';
import useProvideAuth from 'hooks/useSession';
import { useEffect } from 'react';

interface IMainSection extends ReturnType<typeof mapStateToProps> {
  chatId: string;
  toggleChatSettings: typeof toggleChatSettings;
}

function MainSection(props: IMainSection) {
  const { chatId, ws, toggle, toggleChatSettings } = props;
  const { auth } = useProvideAuth();
  const {
    base: {
      default: { color, offColor },
      form: { background },
    },
  } = useThemeColors();
  const { data, refetch } = useGetChatListQuery({ variables: { auth } });
  const { getAllChats } = data || {};
  if (getAllChats?.__typename === 'Error') throw new Error(getAllChats.message);

  useEffect(() => {
    ws.ws?.on('friend_request', () => {
      refetch();
    });
    return () => {
      ws.ws?.off('friend_request');
    };
  }, []);

  return (
    <VStack
      mr="-0.5rem !important"
      ml="-0.5rem !important"
      w={{ base: !toggle.toggleMobileNav ? 0 : '102%', xl: '50%', '2xl': '40%' }}
      h="100vh"
      pos={{ base: 'absolute', lg: 'relative' }}
      bg={background}
      borderRight={useColorModeValue('1px solid rgba(0, 0, 0, 0.1)', '1px solid rgba(255,255,255, 0.1)')}
      textAlign="center"
      overflow="hidden"
      zIndex="20"
      align="center"
      justifyItems="center"
      title="main_section"
    >
      <FindFriends />
      <VStack overflow="auto" w="94%" h="100vh">
        <Slide style={{ zIndex: 10, width: '100%' }} direction="left" in={toggle.toggleChatSettings}>
          <VStack
            w={{ base: !toggle.toggleMobileNav ? 0 : '102%', xl: '50%', '2xl': '35%' }}
            h="100vh"
            top={0}
            left={0}
            transition="0.34s"
            zIndex={11}
            bg={background}
            color="white"
            p={0}
          >
            <Flex ml="2rem" align="center" pos="relative" w="95%" m="1rem" px="1rem">
              <CloseButton
                size="lg"
                className={cx(css`
                  width: 3rem;
                  height: 3rem;
                  cursor: pointer;
                  right: 0;
                  margin-top: 2.5rem;
                  color: ${color};
                  position: absolute;
                  color: ${color};
                  background: ${offColor};
                  z-index: 9999;
                `)}
                onClick={() => {
                  toggleChatSettings(!toggle.toggleChatSettings);
                }}
              />
            </Flex>
            <ChatSettings chatId={chatId} />
          </VStack>
        </Slide>
        {getAllChats?.res.map((item, index) => (
          <ActiveChats key={index} {...item} chatId={chatId} />
        ))}
      </VStack>
    </VStack>
  );
}

const mapStateToProps = (state: STATE) => ({
  toggle: state.toggle,
  ws: state.ws,
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  toggleChatSettings: bindActionCreators(toggleChatSettings, dispatch),
});

export default connect(mapStateToProps, mapDispatchToProps)(MainSection);
