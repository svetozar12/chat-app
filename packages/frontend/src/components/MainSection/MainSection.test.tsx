import "@testing-library/jest-dom";
import MainSection from "./MainSection";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { initialState } from "../../services/redux/reducer/setReducer/setReducer";
import { initialState as saveInputState } from "../../services/redux/reducer/save_inputReducer/save_inputReducer";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);
const store = mockStore({
  setReducer: initialState,
  saveInputReducer: saveInputState,
});

jest.mock("next/router", () => ({
  useRouter() {
    return {
      prefetch: () => null,
    };
  },
}));

const setupRender = () => {
  const component = render(
    <Provider store={store}>
      <MainSection chatRooms={[{ _id: "31231312", members: ["ivan", "greg"] }]} chatId="3213123123" />
    </Provider>,
  );
  return component;
};

describe("Render connected React-redux page", () => {
  beforeEach(() => {
    setupRender();
  });
  it("should create snapshot for <MainSection/>", () => {
    expect(
      renderer
        .create(
          <Provider store={store}>
            <MainSection chatRooms={[{ _id: "31231312", members: ["ivan", "greg"] }]} chatId="3213123123" />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });

  it("should render <MainSection/>", () => {
    const renderedComponent = screen.getByTitle("main_section");
    expect(renderedComponent).toBeInTheDocument();
  });
});
