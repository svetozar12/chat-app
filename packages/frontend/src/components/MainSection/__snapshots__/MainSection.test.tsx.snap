// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <MainSection/> 1`] = `
<section
  className="css-1alui4b"
  title="main_section"
>
  <form
    className="css-zv9gyt"
    onSubmit={[Function]}
  >
    <div
      className="css-1k4crzf"
    />
    <div
      className="css-1s4m24a"
    >
      <div
        className="flex"
      >
        <svg
          className="css-9hnv64"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 496 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
          />
        </svg>
        <h1
          style={
            Object {
              "margin": "0 0 0 1rem ",
              "whiteSpace": "nowrap",
            }
          }
        >
          Chats
        </h1>
      </div>
      <div
        className="flex find_friends_icons"
      >
        <div
          className="flex notifications"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M440.08 341.31c-1.66-2-3.29-4-4.89-5.93-22-26.61-35.31-42.67-35.31-118 0-39-9.33-71-27.72-95-13.56-17.73-31.89-31.18-56.05-41.12a3 3 0 01-.82-.67C306.6 51.49 282.82 32 256 32s-50.59 19.49-59.28 48.56a3.13 3.13 0 01-.81.65c-56.38 23.21-83.78 67.74-83.78 136.14 0 75.36-13.29 91.42-35.31 118-1.6 1.93-3.23 3.89-4.89 5.93a35.16 35.16 0 00-4.65 37.62c6.17 13 19.32 21.07 34.33 21.07H410.5c14.94 0 28-8.06 34.19-21a35.17 35.17 0 00-4.61-37.66zM256 480a80.06 80.06 0 0070.44-42.13 4 4 0 00-3.54-5.87H189.12a4 4 0 00-3.55 5.87A80.06 80.06 0 00256 480z"
            />
          </svg>
        </div>
        <div
          className="flex"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M892 772h-80v-80c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v80h-80c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h80v80c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-80h80c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM373.5 498.4c-.9-8.7-1.4-17.5-1.4-26.4 0-15.9 1.5-31.4 4.3-46.5.7-3.6-1.2-7.3-4.5-8.8-13.6-6.1-26.1-14.5-36.9-25.1a127.54 127.54 0 0 1-38.7-95.4c.9-32.1 13.8-62.6 36.3-85.6 24.7-25.3 57.9-39.1 93.2-38.7 31.9.3 62.7 12.6 86 34.4 7.9 7.4 14.7 15.6 20.4 24.4 2 3.1 5.9 4.4 9.3 3.2 17.6-6.1 36.2-10.4 55.3-12.4 5.6-.6 8.8-6.6 6.3-11.6-32.5-64.3-98.9-108.7-175.7-109.9-110.8-1.7-203.2 89.2-203.2 200 0 62.8 28.9 118.8 74.2 155.5-31.8 14.7-61.1 35-86.5 60.4-54.8 54.7-85.8 126.9-87.8 204a8 8 0 0 0 8 8.2h56.1c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5 29.4-29.4 65.4-49.8 104.7-59.7 3.8-1.1 6.4-4.8 5.9-8.8zM824 472c0-109.4-87.9-198.3-196.9-200C516.3 270.3 424 361.2 424 472c0 62.8 29 118.8 74.2 155.5a300.95 300.95 0 0 0-86.4 60.4C357 742.6 326 814.8 324 891.8a8 8 0 0 0 8 8.2h56c4.3 0 7.9-3.4 8-7.7 1.9-58 25.4-112.3 66.7-153.5C505.8 695.7 563 672 624 672c110.4 0 200-89.5 200-200zm-109.5 90.5C690.3 586.7 658.2 600 624 600s-66.3-13.3-90.5-37.5a127.26 127.26 0 0 1-37.5-91.8c.3-32.8 13.4-64.5 36.3-88 24-24.6 56.1-38.3 90.4-38.7 33.9-.3 66.8 12.9 91 36.6 24.8 24.3 38.4 56.8 38.4 91.4-.1 34.2-13.4 66.3-37.6 90.5z"
            />
          </svg>
        </div>
        <div
          className="flex"
          style={
            Object {
              "cursor": "pointer",
              "position": "relative",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 9.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="flex"
      style={
        Object {
          "position": "relative",
          "width": "95%",
        }
      }
    >
      <div
        className="css-16jj1i0"
      >
        <svg
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "black",
              "cursor": "pointer",
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
          />
        </svg>
        <input
          className="css-1xg5m5i"
          onChange={[Function]}
          placeholder="Search for chats"
          type="search"
          value=""
        />
      </div>
    </div>
  </form>
  <div
    className="flex "
    style={
      Object {
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "flex-start",
        "overflow": "auto",
        "width": "95%",
      }
    }
  >
    <div
      className="flex css-viynhm"
    >
      <div
        className="flex css-1slvo6y"
      >
        <svg
          className="css-2ulnmz"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3,3 L21,21 M3,21 L21,3"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
      </div>
      <div
        className="flex css-w2h0t4"
      >
        <h1
          className="css-zt1gbb"
        >
          Members in chat
        </h1>
      </div>
    </div>
    <div
      className="css-1ucxuqo"
      data-testid="chat"
      onClick={[Function]}
    >
      <div
        className="css-1lymioy"
      >
        <section
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <section
            data-testid="avatar"
          >
            <div>
              <div
                style={
                  Object {
                    "marginRight": "1rem",
                  }
                }
              >
                <svg
                  className="css-nkhozk css-boia29"
                  data-testid="single_avatar"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                      "height": undefined,
                      "width": undefined,
                    }
                  }
                  viewBox="0 0 496 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                  />
                </svg>
              </div>
            </div>
          </section>
          <div
            className="css-1c5u6r2"
          >
            <div
              className="css-19dbfyd"
            />
            <p
              className="css-1p5u1lq"
            >
              Last message...
            </p>
          </div>
        </section>
      </div>
    </div>
  </div>
</section>
`;
